### PR TITLE
chore: deprecate chromeWebSecurity

### LINKED ISSUE #19892: [Multi-domain]: Mark chromeWebSecurity as deprecated
### What would you like?

When the experimental flag is enabled, mark chromeWebSecurity as deprecated in config.

### Why is this needed?

ideally chromeWebSecurity would not be set to false when testing.

### PR DESCRIPTION
**Pull request by mschile**
- Closes #19892

### User facing changelog

The `chromeWebSecurity` configuration option has been deprecated and will be removed in a future release. We encourage you to use `cy.origin` instead.

### Additional details

With the addition of `cy.origin`, the `chromeWebSecurity` configuration option is no longer needed to allow users to navigate cross-origin. As such, this change deprecates the `chromeWebSecurity` option and notifies the user via a terminal warning that the option has been deprecated.

### How has the user experience changed?

Screen Shot 2022-05-11 at 10 47 29 AM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by mjhenkes — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]

**Review by lmiller1990 — COMMENTED** [review]
Just a question, isn't there some use cases where `cy.origin` is limited and `CHROME_WEB_SECURITY` might still be useful?

File: packages/driver/cypress/integration/commands/navigation_spec.js [hunk 1]
```diff
@@ -2177,11 +2177,14 @@ describe('src/cy/commands/navigation', () => {
               > ${error}\n
             Before the page load, you were bound to the origin policy:\n
               > http://localhost:3500\n
+            In order to navigate to a different origin, you can enable the \`experimentalSessionAndOrigin\` flag and use \`cy.origin()\`:\n
+            \`cy.get('.goToAnotherOrigin').click()\`
+            \`cy.origin('http://localhost:3500', () => {\`
+            \`  <commands targeting http://localhost:3500 go here>\`
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/commands/navigation_spec.js:2183]: I'm not sure I've seen bracket syntax in our error messages. have we used JS comment syntax at all?

**Comment by emilyrohrbough** [comment]
We are able to surface warnings in the desktop-gui. We do it with the baseUrl when the server has not been started which would likely cause test failures. Can we surface this warning as well so users are aware of this deprecation?

Screen Shot 2022-05-12 at 8 09 14 AM

File: cli/schema/cypress.schema.json [hunk 2]
```diff
@@ -181,7 +181,7 @@
         "chromeWebSecurity": {
           "type": "boolean",
           "default": true,
-          "description": "Whether Chrome Web Security for same-origin policy and insecure mixed content is enabled. Read more about this at https://on.cypress.io/web-security"
+          "description": "DEPRECATED: Please use cy.origin() instead. Whether Chrome Web Security for same-origin policy and insecure mixed content is enabled. Read more about this at https://on.cypress.io/web-security"
```
Comment (emilyrohrbough) [inline cli/schema/cypress.schema.json:184]: We need to add the deprecation to the cypress types as well: https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts#L2805

**Review by emilyrohrbough — CHANGES_REQUESTED** [review]
Can you update the summary and/or add details on the impact this change will have on consumers?

We also need to open a corresponding cypress-docs pr to mark this as deprecated and provide details on the path forward for those leveraging this.

File: packages/driver/cypress/integration/commands/navigation_spec.js (hunk 1 above)
Comment (mschile) [reply packages/driver/cypress/integration/commands/navigation_spec.js:2183]: This is the syntax we've used for the other cross origin error messages. I don't see comment syntax being used in the other error messages. The only other message I see uses `...`:
```
describe('{{title}}', { retries: {{numRetries}} }, () => {
  ...
})
```

**Comment by mschile** [comment]
Closing PR.

We've decided to wait on deprecating `chromeWebSecurity` until we have `iframe` support for `cy.origin`.

**Pull request closed without merging**